src/Features/LanguageServer/Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs

```diff
@@ -38,11 +39,8 @@ protected override VSInternalWorkspaceDiagnosticReport CreateReport(TextDocument
                 Identifier = WorkspaceDiagnosticIdentifier,
             };
 
-        protected override IProgress<VSInternalWorkspaceDiagnosticReport[]>? GetProgress(VSInternalWorkspaceDiagnosticsParams diagnosticsParams)
-            => diagnosticsParams.PartialResultToken;
-
-        protected override VSInternalDiagnosticParams[]? GetPreviousResults(VSInternalWorkspaceDiagnosticsParams diagnosticsParams)
-            => diagnosticsParams.PreviousResults;
+        protected override PreviousResult[]? GetPreviousResults(VSInternalWorkspaceDiagnosticsParams diagnosticsParams)
```

Comment:
in general, unless you need an array (e.g. for json purposes) can we use things like ImmutableArray internally to make the semantics of mutation clear?